lib/stdlib/src/erl_lint.erl

```diff
@@ -676,6 +678,9 @@ start(File, Opts) ->
          {singleton_typevar,
           bool_option(warn_singleton_typevar, nowarn_singleton_typevar,
                       true, Opts)},
+        {multiple_definitions_of_type,
```

Comment:
Assuming this PR goes into `master`, should it really be possible to turn off the check multiple definitions of a type?

Reply:
That depends on whether this is considered a bug and we force clients to update their code bases.
I would imagine that some projects may have plenty of these duplicate name definitions and, if so, maybe it is a good thing to give them some way to eventually remove these issues? At least for 1 OTP release if we consider this a breaking change?

Reply:
Yes, you are right. It is probably better to make it possible to turn off in OTP 27.